### PR TITLE
fix(amplify-nodejs-function-runtime-provider): fix lambda error format

### LINKED ISSUE #5553: Best way to develop code to run for Lambda's, current setup is painfully slow and manual
>**Note**: If your question is regarding the AWS Amplify Console service, please log it in the 
[official AWS Amplify Console forum](https://forums.aws.amazon.com/forum.jspa?forumID=314&start=0)

**Which Category is your question related to?**

I'm after knowing if there is a better way to develop my code for running in a Function/Lambda. Currently (after initially creating the Function via Amplify) I'm manually editing the code in the browser editor for the Lambda itself. I then trigger the Lambda by uploading a file to S3 (its an import so this is how it will be run anyway). That loop though is slow going and totally kills any flow.

I've tried mocking the Function locally but if there are any errors or typos in my code it just hangs and won't give any details so the only way I've found to get anything sensible out is to edit it on live/dev directly.

Deploying a Lambda Layer for the shared/core code did wonders for making development easier, I wonder if there is something else I'm missing that might make this more sane and speedy?

**Amplify CLI Version** 

You can use `amplify -v` to check the amplify cli version on your system

4.29.4

**What AWS Services are you utilizing?**

Functions/Lambda

**Provide additional details e.g. code snippets**

### PR DESCRIPTION
**Pull request by SwaySway**
*Issue #, if available:*
fix #5553
*Description of changes:*
return proper error message and stack if a string error is provided return string error as is

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-nodejs-function-runtime-provider/src/utils/execute.ts [hunk 1]
```diff
@@ -95,14 +95,12 @@ process.on('message', async options => {
     process.stdout.write('\n');
     process.stdout.write(JSON.stringify({ result, error: null }));
   } catch (error) {
+    let lambdaError = typeof error === 'string' ? { message: 'Unkown Error' } : { message: error.message, stack: error.stack };
```
Comment (edwardfoyle) [inline packages/amplify-nodejs-function-runtime-provider/src/utils/execute.ts:98]: typo: `Unknown Error`

File: packages/amplify-nodejs-function-runtime-provider/src/__tests__/utils/__snapshots__/execute.test.ts.snap [hunk 2]
```diff
@@ -565,4 +565,8 @@ exports[`execute callback handler should fail when returning an error 1`] = `[Er
 
 exports[`execute callback handler should succeed with a returned value 1`] = `"foo"`;
 
+exports[`execute failed execution should return the proper error 1`] = `[ReferenceError: undedvar is not defined]`;
```
Comment (edwardfoyle) [inline packages/amplify-nodejs-function-runtime-provider/src/__tests__/utils/__snapshots__/execute.test.ts.snap:568]: shouldn't this error message also include the stacktrace?

**Review by edwardfoyle — COMMENTED** [review]
one typo and one clarification question

File: packages/amplify-nodejs-function-runtime-provider/src/__tests__/utils/__snapshots__/execute.test.ts.snap (hunk 2 above)
Comment (SwaySway) [reply packages/amplify-nodejs-function-runtime-provider/src/__tests__/utils/__snapshots__/execute.test.ts.snap]: Updated stacktrace is included added a different test suite for this

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**